### PR TITLE
Improve roll.py

### PR DESCRIPTION
**Pull request by abarth**
* Make the paths absolute
 * Using filter to remove tonic from mojo/dart/embedder/BUILD.gn instead of a
   patch

### CODE REVIEW

File: sky/tools/roll/patch.py [hunk 1]
```diff
@@ -14,6 +14,10 @@ def patch_and_filter(dest_dir, relative_patches_dir):
       lambda line: not "gyp_environment" in line)
   utils.commit("filter gyp_environment out of build/landmines.py")
 
+  utils.filter_file("mojo/dart/embedder/BUILD.gn",
```
Comment (jamesr) [inline sky/tools/roll/patch.py]: you wanna nuke the patchfile for dart/embedder/ too, right?

Comment (abarth) [inline sky/tools/roll/patch.py]: I did that as part of the roll because otherwise it would cause this script to fail.

**Comment by jamesr** [comment]
How do the #include tonic/... lines in mojo/dart/embedder/ pass gn check in the flutter repo?

**Comment by jamesr** [comment]
Looks like //mojo/dart/embedder:dart_controller_no_snapshot #includes the headers from sky/engine/tonic/... but it doesn't declare any dependency, but since nothing in the tree says that it owns "tonic/dart_converter.h" gn check just ignores those lines.

**Comment by jamesr** [comment]
lgtm

**Pull request merged**